code/modules/food/icecreamvat.dm

```diff
@@ -12,51 +12,73 @@
 /obj/machinery/cooking/icemachine
 	name = "Cream-Master Deluxe"
 	icon_state = "icecream_vat"
-
 	var/obj/item/weapon/reagent_containers/glass/beaker = null
+	var/datum/reagents/to_add
+	var/soda
+	var/alcohol
 
 /obj/machinery/cooking/icemachine/New()
-	src.reagents = new/datum/reagents(500)
-	src.reagents.my_atom = src
-	return ..()
+	create_reagents(500)
+	flags |= NOREACT
+	to_add = new/datum/reagents(30)
+	to_add.my_atom = src
+	soda = pick(COLA,DR_GIBB,SPACE_UP,SPACEMOUNTAINWIND)
+	alcohol = pick(KAHLUA,VODKA,RUM,GIN)
+	generate_icecream_reagents(51)
+	..()
+
+/obj/machinery/cooking/icemachine/Destroy()
+	qdel(beaker)
+	beaker = null
+	qdel(to_add)
+	to_add = null
+	..()
 
 // Utilities ///////////////////////////////////////////////////
 
+/obj/machinery/cooking/icemachine/proc/generate_icecream_reagents(var/reagent_total)
+	reagents.add_reagent(soda, min(reagent_total-reagents.get_reagent_amount(soda), reagent_total))
+	reagents.add_reagent(alcohol, min(reagent_total-reagents.get_reagent_amount(alcohol), reagent_total))
+	reagents.add_reagent(CREAM, min(reagent_total-reagents.get_reagent_amount(CREAM), reagent_total))
+	reagents.add_reagent(WATER, min(reagent_total-reagents.get_reagent_amount(WATER), reagent_total))
```

Comment:
this should be ICE

Reply:
Maybe so, but ice dispensing is broken right now. #17029